### PR TITLE
Fix QProcess failing to launch autoscoper.exe on Windows

### PR DESCRIPTION
**Pull request by NicerNewerCar**
* Removed usage of `QProcess.start(executablePath)` in favor of:
```python
QProcess.setProgram(executablePath)
QProcess.start()
```
* Fixed bug with AutoscoperM refusing to reconnect to Autoscoper server if the connection was severed outside of the Slicer environment (ie. Autoscoper crashed and you tried to relaunch it from Slicer)
* ~~Updated `.gitignore` to include build directories~~
* Closes Issues #15 and #18

### CODE REVIEW

**Comment by amymmorton** [comment]
I don't think I can test until I build (after the slicer build from source is complete)- unless you want to commit and we test the extension directly?

**Comment by jcfr** [comment]
@amymmorton you could modify the file locally based on the content proposed in the pull-request and restart Slicer.

File: AutoscoperM/AutoscoperM.py [hunk 1]
```diff
@@ -236,8 +236,10 @@ def onApplyButton(self):
         import sys  # noqa: F401
 
         if sys.platform == "darwin":
-            slicer.util.messageBox("AutoscoperM is not supported on this platform.<br>"
-                "See <a href='https://autoscoperm.slicer.org'>https://autoscoperm.slicer.org</a> for details.")
+            slicer.util.messageBox(
+                "AutoscoperM is not supported on this platform.<br>"
+                "See <a href='https://autoscoperm.slicer.org'>https://autoscoperm.slicer.org</a> for details."
+            )
```
Comment (jcfr) [inline AutoscoperM/AutoscoperM.py:242]: Since the line-length associated with `black` is only specified in .pre-commit-config.yaml, I suspect the your local development environment has a lower default for the line length.

https://github.com/BrownBiomechanics/SlicerAutoscoperM/blob/1a840deeb03a5f7936a26653e60edf2a58bba542/.pre-commit-config.yaml#L5-L6

File: AutoscoperM/AutoscoperM.py [hunk 2]
```diff
@@ -366,10 +364,7 @@ def stopAutoscoper(self, force=True):
         if self.AutoscoperSocket:
             self.disconnectFromAutoscoper()
 
-        if force:
-            self.AutoscoperProcess.kill()
-        else:
-            self.AutoscoperProcess.terminate()
+        self.AutoscoperProcess.kill()
```
Comment (jcfr) [inline AutoscoperM/AutoscoperM.py:367]: Assuming this change and the one above checking if the connection is already established are improvement, they should be done in a dedicated commit.

If the changes are part of the solution, the commit message should explain it.

Comment (jcfr) [reply AutoscoperM/AutoscoperM.py:367]: > Fixed bug with AutoscoperM refusing to reconnect to Autoscoper server if the connection was severed outside of the Slicer environment (ie. Autoscoper crashed and you tried to relaunch it from Slicer)

In other words, this should be fixed in its own commit.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,2 @@
 AutoscoperM/__pycache__/*
-
+build*/
```
Comment (jcfr) [inline .gitignore:2]: It is recommended to do out-of-source build

Comment (jcfr) [reply .gitignore:2]: This is relevant:
* https://github.com/BrownBiomechanics/SlicerAutoscoperM/pull/22

**Comment by amymmorton** [comment]
I attempted to make local changes (C:\Users\amorton1\AppData\Local\slicer.org\Slicer 5.3.0-2023-03-27\slicer.org\Extensions-31693\SlicerAutoscoperM\lib\Slicer-5.3\qt-scripted-modules\AutoscoperM.py)
 lines 347-353:     

with changeCurrentDir(executableDirectory):
            logging.info("Starting Autoscoper %s" % executablePath)
            self.AutoscoperProcess.setProgram(executablePath)
            self.AutoscoperProcess.start()
            self.AutoscoperProcess.waitForStarted()

        slicer.app.processEvents()



After slicer restart- the app launches successfully:

[FD] Initializing OpenGL...
[FD] Initializing OpenCL-OpenGL interoperability...
[FD] Initializing OpenGL...
[FD] Initializing OpenCL-OpenGL interoperability...
[FD] New Client is Connected...



Trials open successfully

**Comment by amymmorton** [comment]
machine restart did not change the error
